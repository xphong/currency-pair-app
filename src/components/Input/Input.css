.input {
  padding: 6px 24px;
  margin: 6px 0;
  width: 100%;
  font-size: 14px;
}
