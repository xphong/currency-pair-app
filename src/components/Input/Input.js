import React from 'react';
import PropTypes from 'prop-types';

import styles from './Input.css';

const Input = props => {
  return <input className={styles.input} {...props} />
}

Input.propTypes = {
  type: PropTypes.string.isRequired
};

Input.defaultProps = {
  type: 'text'
};

export default Input;
