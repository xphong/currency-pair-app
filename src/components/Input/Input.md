### Default Style

```js
<Input />
```

### Radio

```js
<Input type="radio" />
```

### Checkbox

```js
<Input type="checkbox" />
```
