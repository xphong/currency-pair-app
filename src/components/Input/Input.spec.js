import React from 'react';

import Input from './Input';

describe('Input Component', () => {
  it('should render Input', () => {
    const wrapper = shallow(<Input type="text" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render Input with props', () => {
    const props = {
      value: 'Goku',
      name: 'Kakarot'
    }
    const wrapper = shallow(<Input type="text" {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render Input with type number', () => {
    const wrapper = shallow(<Input type="number" />);

    expect(wrapper).toMatchSnapshot();
  });
});
