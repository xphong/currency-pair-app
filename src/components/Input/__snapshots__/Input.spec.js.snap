// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render Input 1`] = `
<input
  className="input"
  type="text"
/>
`;

exports[`Input Component should render Input with props 1`] = `
<input
  className="input"
  name="Kakarot"
  type="text"
  value="Goku"
/>
`;

exports[`Input Component should render Input with type number 1`] = `
<input
  className="input"
  type="number"
/>
`;
